base/cvd/cuttlefish/host/libs/config/fetched_archive.cc

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright (C) 2025 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cuttlefish/host/libs/config/fetched_archive.h"
+
+#include <stddef.h>
+
+#include <map>
+#include <optional>
+#include <ostream>
+#include <set>
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include "absl/strings/match.h"
+#include "absl/strings/strip.h"
+#include "fmt/ostream.h"
+#include "fmt/ranges.h"
+
+#include "cuttlefish/common/libs/utils/files.h"
+#include "cuttlefish/common/libs/utils/result.h"
+#include "cuttlefish/host/libs/config/fetcher_config.h"
+#include "cuttlefish/host/libs/config/file_source.h"
+#include "cuttlefish/host/libs/zip/libzip_cc/archive.h"
+#include "cuttlefish/host/libs/zip/zip_file.h"
+
+namespace cuttlefish {
+
+Result<FetchedArchive> FetchedArchive::Create(
+    const FetcherConfig& fetcher_config, FileSource source,
+    std::string_view archive) {
+  std::optional<ReadableZip> zip_file;
+  std::set<std::string> members;
+  std::map<std::string, std::string> extracted_members;
+
+  for (const auto& [path, member] : fetcher_config.get_cvd_files()) {
+    if (member.source != source) {
+      continue;
+    }
+
+    if (absl::EndsWith(path, archive) && absl::EndsWith(archive, ".zip")) {
```

Comment:
The `EndsWith` check could pass for a path ending in `foobar.zip`, even if archive was `bar.zip`. Consider using `android::base::Basename` and equality instead, or `path == archive || absl::EndsWith(path, "/"+archive)`